### PR TITLE
[CGP-364] [Tests] [Type checker] Added tests for 'typeOfBuiltinName'

### PR DESCRIPTION
**Pull request by effectfully**
Please check the golden files.

### CODE REVIEW

File: language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs [hunk 1]
```diff
@@ -49,7 +50,7 @@ data BuiltinName = AddInteger
                  | TxHash
                  | BlockNum
                  | BlockTime
-                 deriving (Show, Eq, Ord, Generic, NFData, Lift)
```
Comment (effectfully) [inline language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs]: Do we plan to remove the last two built-ins?

Comment (vmchale) [reply language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs]: BlockTime is no longer in the spec so you can safely remove it.

Comment (effectfully) [reply language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs]: What about `BlockNum`?

Comment (mchakravarty) [reply language-plutus-core/src/Language/PlutusCore/Lexer/Type.hs]: I think, we decided to keep `blocknum` as a proxy for blockchain information provided to validator code. (In concrete uses of PLC on the blockchain, there will be further builtins, but they depend on the particular chain.)

**Comment by effectfully** [comment]
I'd like to merge this without removing `BlockTime`, because if we decided to keep `BlockNum`, then I need to update some stuff and that work on built-ins fits into a separate PR (the next one).

**Pull request merged**